### DISCUSSION TITLE
Understanding the use of `clang-tidy`

### DISCUSSION THREAD

**Discussion by aske-w** [duckdb/duckdb#8789] [Q&A]
Hello. Is the use of `clang-tidy` a fix-as-you-go kind of thing? 

I see it was added in #1382 in February 2021 and a few fixes of rule violations in #7670 but other violations are still reported in my clone of the repository. It is checked out at tag v0.8.1.

**Comment by carlopi** [comment]
Idea is that CI (CodeQuality.yml) will enforce that PRs will have to pass (as a whole) the clang-tidy rules.
Over time `clang-tidy` will change slightly, so it's expected that a given version of the codebase will pass on a given version of `clang-tidy`, hard to say more than that.

Advice on improving process are welcome.

**Reply by aske-w** [reply]
So right now the CodeQuality.yml CI workflow does not fail even if `clang-tidy` finds violations, but it is the long term goal to make the code base adhere to the `clang-tidy` rules and make the workflow fail on any violation?

**Reply by carlopi** [reply]
No, currently one of the jobs in CodeQuality.yml is running clang-tidy (with a specific rule-set & version): https://github.com/duckdb/duckdb/blob/main/.github/workflows/CodeQuality.yml#L68

**Reply by carlopi** [reply]
Can you show an example of violation you can find on `main`?

**Reply by aske-w** [reply]
I get `error: performing an implicit widening conversion to type 'size_type' (aka 'unsigned long') of a multiplication performed in type 'int' [bugprone-implicit-widening-of-multiplication-result,-warnings-as-errors]` on several lines in `query_profiler.cpp` when running `make tidy-check` locally: https://github.com/duckdb/duckdb/blob/main/src/main/query_profiler.cpp#L489C19-L492
